### ISSUE TITLE
Customer.retrievePaymentMethod() is broken

### ISSUE THREAD

**Issue by alan-czajkowski** [stripe/stripe-java#1581]
### Describe the bug

unable to retrieve a customer's payment method using: `Customer.retrievePaymentMethod()`

the `Customer.retrievePaymentMethod()` method is broken: when the URL parameters are inserted by `String.format()` it inserts the customer id for both the customer and the payment method instead of inserting the correct payment method id

### To Reproduce

```java
...
Customer customer = ...
customer.retrievePaymentMethod(customer.getId(), ..., ...)
...
```

### Expected behavior

the method should take in the payment method id, not the customer id, in order for the `String.format()` to work correctly

### Code snippets

https://github.com/stripe/stripe-java/blob/master/src/main/java/com/stripe/model/Customer.java#L561-L588

Look at the String.format lines of code, they are not inserting the id's correctly:

```java
  /** Retrieves a PaymentMethod object for a given Customer. */
  public PaymentMethod retrievePaymentMethod(
      String customer, Map<String, Object> params, RequestOptions options) throws StripeException {
    String url =
        ApiResource.fullUrl(
            Stripe.getApiBase(),
            options,
            String.format(
                "/v1/customers/%s/payment_methods/%s",
                ApiResource.urlEncodeId(customer), ApiResource.urlEncodeId(this.getId())));
    return ApiResource.request(
        ApiResource.RequestMethod.GET, url, params, PaymentMethod.class, options);
  }

  /** Retrieves a PaymentMethod object for a given Customer. */
  public PaymentMethod retrievePaymentMethod(
      String customer, CustomerRetrievePaymentMethodParams params, RequestOptions options)
      throws StripeException {
    String url =
        ApiResource.fullUrl(
            Stripe.getApiBase(),
            options,
            String.format(
                "/v1/customers/%s/payment_methods/%s",
                ApiResource.urlEncodeId(customer), ApiResource.urlEncodeId(this.getId())));
    return ApiResource.request(
        ApiResource.RequestMethod.GET, url, params, PaymentMethod.class, options);
  }
```
```


### OS

macOS

### Java version

Java 17

### stripe-java version

v22.23.0

### API version

2022-11-15

### Additional context

_No response_

**Comment by richardm-stripe** [comment]
Hello @alan-czajkowski, happy Friday! Thanks for the report. The fix is released in [v22.23.1](https://github.com/stripe/stripe-java/releases/tag/v22.23.1)

**Comment by alan-czajkowski** [comment]
@richardm-stripe thank you for the quick turn-around
